### PR TITLE
Diff Cleanup

### PR DESCRIPTION
**Pull request by dcy2003**
This diff is massive and **I would highly encourage someone to pull down my branch and take it for a ride locally prior to merging...**

Contains fixes for #591.  Specifically formatter plugin was brought up to date and templates were updated to Mars format with same values.  Formatting now appears to work correctly from command line when not skipped (which by the way, is now done via `-Dformatter.skip=true` instead of `-DskipFormat=true`).

Contains a massive format of the codebase to put files in this format.

Also normalizes all line endings via dos2unix which was not done correctly with #755

### CODE REVIEW

File: analytics/api/src/main/java/mil/nga/giat/geowave/analytic/GeometryDataSetGenerator.java [hunk 1]
```diff
@@ -147,9 +147,11 @@ public Geometry getBoundingRegion() {
...
 		final double[] max = new double[range.length];
 		for (int i = 0; i < range.length; i++) {
-			min[i] = Math.max(
-					minAxis[i] + (minCenterDistanceFactor * (rand.nextInt(Integer.MAX_VALUE) % (range[i] / minCenterDistanceFactor))),
-					minAxis[i]);
+			min[i] = Math
+					.max(
+							minAxis[i]
+									+ (minCenterDistanceFactor * (rand.nextInt(Integer.MAX_VALUE) % (range[i] / minCenterDistanceFactor))),
+							minAxis[i]);
```
Comment (datasedai) [inline analytics/api/src/main/java/mil/nga/giat/geowave/analytic/GeometryDataSetGenerator.java]: This looks really bizarre ... is this the correct formatting?

File: analytics/api/src/main/java/mil/nga/giat/geowave/analytic/clustering/NeighborData.java [hunk 2]
```diff
@@ -102,6 +102,7 @@ public int compareTo(
 
 	@Override
 	public String toString() {
-		return (id == null ? "" : Hex.encodeHexString(id.getBytes()) + ":") + element.toString() + "(" + this.distance + ")";
+		return (id == null ? "" : Hex.encodeHexString(id.getBytes()) + ":") + element.toString() + "(" + this.distance
+				+ ")";
```
Comment (datasedai) [inline analytics/api/src/main/java/mil/nga/giat/geowave/analytic/clustering/NeighborData.java]: ...why?  I can't see any possible benefit from this being on the next line ... is this correct?

File: analytics/api/src/main/java/mil/nga/giat/geowave/analytic/partitioner/AdapterBasedPartitioner.java [hunk 3]
```diff
@@ -108,7 +108,9 @@ protected NumericDataHolder getNumericData(
 				entry.data,
 				getIndex().getIndexModel());
 		final double[] thetas = getDistancePerDimension();
-		final MultiDimensionalNumericData primaryData = encoding.getNumericData(getIndex().getIndexModel().getDimensions());
+		final MultiDimensionalNumericData primaryData = encoding.getNumericData(getIndex()
+				.getIndexModel()
```
Comment (datasedai) [inline analytics/api/src/main/java/mil/nga/giat/geowave/analytic/partitioner/AdapterBasedPartitioner.java]: How did this code change?  I thought this is a formatting only PR?

File: analytics/mapreduce/src/main/java/mil/nga/giat/geowave/analytic/mapreduce/kde/KDEJobRunner.java [hunk 4]
```diff
@@ -205,7 +209,9 @@ public int runJob()
 			statsReducer.setMapperClass(IdentityMapper.class);
 			statsReducer.setPartitionerClass(getJob2Partitioner());
 			statsReducer.setReducerClass(getJob2Reducer());
-			statsReducer.setNumReduceTasks(getJob2NumReducers((kdeCommandLineOptions.getMaxLevel() - kdeCommandLineOptions.getMinLevel()) + 1));
+			statsReducer
+					.setNumReduceTasks(getJob2NumReducers((kdeCommandLineOptions.getMaxLevel() - kdeCommandLineOptions
```
Comment (datasedai) [inline analytics/mapreduce/src/main/java/mil/nga/giat/geowave/analytic/mapreduce/kde/KDEJobRunner.java]: Is this correct?  Seems like odd formatting

**Comment by datasedai** [comment]
Reviewed, didn't see anything major, but could you update the docs in "Building -> Tools" to update the skipFormat command that we always copy/paste always?

File: analytics/api/src/main/java/mil/nga/giat/geowave/analytic/clustering/NeighborData.java (hunk 2 above)
Comment (dcy2003) [inline analytics/api/src/main/java/mil/nga/giat/geowave/analytic/clustering/NeighborData.java]: The formatter defines the following, which is why this one occurs:
`<setting id="org.eclipse.jdt.core.formatter.lineSplit" value="120"/>`

File: analytics/mapreduce/src/main/java/mil/nga/giat/geowave/analytic/mapreduce/kde/KDEJobRunner.java (hunk 4 above)
Comment (dcy2003) [inline analytics/mapreduce/src/main/java/mil/nga/giat/geowave/analytic/mapreduce/kde/KDEJobRunner.java]: Same as above, this one is being caused by:
`<setting id="org.eclipse.jdt.core.formatter.lineSplit" value="120"/>`

File: analytics/api/src/main/java/mil/nga/giat/geowave/analytic/partitioner/AdapterBasedPartitioner.java (hunk 3 above)
Comment (dcy2003) [inline analytics/api/src/main/java/mil/nga/giat/geowave/analytic/partitioner/AdapterBasedPartitioner.java]: This is a formatting only PR and I verified against master that there is no code change here (this line was already present), not sure why exactly it's showing up in the diff like this.

**Pull request merged**